src/microbe_stage/systems/MucocystSystem.cs

```diff
@@ -37,6 +34,13 @@ protected override void Update(float delta, in Entity entity)
         if (control.MucocystEffectsApplied == wantsMucocyst && !control.MucocystEffectsApplied)
             return;
 
+        // Disable mucocyst if engulfed
+        if (entity.Get<Engulfable>().PhagocytosisStep != PhagocytosisPhase.None)
```

Comment:
I think it would be better to put this code where other special state resetting when engulfed handling is done, because as written this system has to be modified to say that this only processes entities with Engulfable component (and just reads it).

Reply:
I have fixed it now. I'm not sure if this is the proper solution, as the special state resetting is technically already done in 'EngulfedHandlingSystem'. However, it runs after the mucocyst system, so mucocyst is deactivated with a slight delay, causing a flicker.
That could probably be fixed with a `[RunsAfter()]` attribute, but it didn't seem to help.

Reply:
I assume the check to not allow entering mucocyst state when engulfed in the input handling should take care of this.